docs/appnotes/0015-using-tams-in-opentimelineio.md

```diff
@@ -0,0 +1,171 @@
+            "rate": 1000000000.0,
+            "value": 1723124225400000000.0
+        }
+    },
+    "effects": [],
+    "markers": [],
+    "enabled": true,
+    "media_references": {
+        "DEFAULT_MEDIA": {
+            "OTIO_SCHEMA": "ExternalReference.1",
+            "metadata": {},
+            "name": "camera-one.ts",
+            "available_range": {
+                "OTIO_SCHEMA": "TimeRange.1",
+                "duration": {
+                    "OTIO_SCHEMA": "RationalTime.1",
+                    "rate": 1000000000.0,
+                    "value": 372100000000.0
+                },
+                "start_time": {
+                    "OTIO_SCHEMA": "RationalTime.1",
+                    "rate": 1000000000.0,
+                    "value": 1723124086620000000
+                }
+            },
+            "available_image_bounds": null,
+            "target_url": "tamss://tams.example.com/flows/9bb414a5-862c-494f-86ce-8e2720ecc315"
+        }
+    },
+    "active_media_reference_key": "DEFAULT_MEDIA"
+}
+```
+
+To make use of this format, a tool should either use its native TAMS integration to contact the store and fetch the content, or alternatively a Media Linker could be written to download the relevant content to a file (and then another tool used to remap back to references in the store).
+
+## Rendering back to TAMS
+
+TAMS has a basic mechanism for re-using essence in multiple places on multiple Flow timelines, as described in the [Flow and Media Timelines section](../../README.md#flow-and-media-timelines) of the top-level README.
+When a composition is rendered by a tool, this mechanism could be used to make the render process faster and more storage efficient, by identifying areas that can be re-used, while writing new Flow Segments for areas that cannot.
+
+For example, consider a pair of clips (Sources) and a timeline built out of them, with a number of graphics and transitions applied as well.
+Internally, this timeline is represented as a data structure describing clips and effects, for example stored in an OpenTimelineIO composition.
+
+![Input material and basic composition](./images/0015-opentimeline-writeback-composition.png)
+
+Clearly the result of rendering this timeline is a new Source and Flow, and it could be rendered in the usual way, producing new essence that is written to the store.
+However, it could also be rendered by identifying areas of the timeline that copy the inputs directly, and creating Flow references instead.
+In the diagram below Source/Flow `sO/fO` have been created as the output, containing a mix of references to inputs, and new material where references are not possible.
+
+![Render process and new Flows](./images/0015-opentimeline-writeback-render.png)
+
+This makes the render process faster (references are a very "cheap" metadata operation), makes more efficient use of storage, and avoids an additional decode-encode cycle on some of the content.
```

Comment:
(repeating discussions that have been had and the description further below)
and the reason for having those references in TAMS rather than keeping it in an external EDL is that it adds the bits that would otherwise to missing from the Flow that contains the new material segments - it becomes more a published artefact rather than an ephemeral work-in-progress artefact.. It can then be played back by processes that may not have the capabilities to render (parts of) the EDL. Although the other option is to pass a flattened EDL that contains the missing references to other Flows.

Reply:
Good to call this out - I've added a few words to this effect